data_prep/introspector.py

```diff
@@ -39,64 +39,55 @@
 INTROSPECTOR_FUNC_SIG = f'{INTROSPECTOR_ENDPOINT}/function-signature'
 
 
+def _query_introspector(api: str, params: dict, key: str) -> Any:
```

Comment:
Let's avoid using `Any`, because it defeats the purpose of type-checking.
For example, this can be `-> dict` and `return data`.
This function checks `resp.ok`, then each caller checks the value they care about.